lib/workos/api.ex

```diff
@@ -7,12 +7,16 @@ defmodule WorkOS.API do
   Generates the Tesla client used to make requests to WorkOS
   """
   def client(opts \\ []) do
+    auth = WorkOS.api_key(opts)
+    if is_map_key(opts, :access_token) do
```

Comment:
should we be throwing an error if there isn't an `access_token` or are there some endpoints that don't require it?

Reply:
So right now, the only endpoint that would have an access_token is the `get_profile` - all other endpoints would use the API Key if a Bearer Token is needed.

https://workos.com/docs/reference/sso/profile/get-user-profile

I tried to support that here, but I may have missed some logic.

Reply:
Isn't that what the `api_key` function is doing already? Looks like it will return the access_token from the options, or get the environment variable if nothing is passed in

https://github.com/workos/workos-elixir/blob/main/lib/workos.ex#L6-L8

Reply:
The API key is used as the Authorization Bearer Token for most endpoints, for example, we use it for Connections API: https://workos.com/docs/reference/sso/connection/get

The API Key is also the client secret when users hit the token endpoint: https://workos.com/docs/reference/sso/profile/get-profile-and-token

However, the Get User Profile endpoint uses the access_token received from the Token endpoint as the Authorization Bearer Token. This is more in line with a true OAuth 2 authorization code grant flow. It's only needed for clients that have to use this flow, and call a user info endpoint after receiving the access token.

Reply:
ah ok that makes sens